### ISSUE TITLE
Fatal errors while executing functional tests

### ISSUE THREAD

**Issue by markwatney2016** [TYPO3/testing-framework#6]
There may be a problem while executing the functional tests of custom extensions (using `typo3/testing-framework`) in TYPO3 CMS projects? In our projects it seems as if the database credentials set via environment variables are correcty retrieved via `$localConfiguration['DB'] = $testbase->getOriginalDatabaseSettingsFromEnvironmentOrLocalConfiguration();` in `TYPO3\TestingFramework\Core\Functional\FunctionalTestCase` but are not used when executing a test: there the database credentials set in the project’s `Localconfiguration.php` seem to be used.

We use both the current `dev-master` of TYPO3 CMS and the current `dev-master` of `typo3/testing-framework`.

```
There were 29 errors:

1) ACME\Base\Tests\Functional\Service\SettingsServiceTest::canReturnFrameworkConfiguration
Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: No such file or directory

/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:103
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:145
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/Driver.php:43
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:64
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:173
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1398
/Volumes/Projects/MyProject/Web/vendor/typo3/testing-framework/Classes/Core/Testbase.php:503
/Volumes/Projects/MyProject/Web/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:274
/Volumes/Projects/MyProject/Web/typo3conf/ext/acme_base/Tests/Functional/Service/SettingsServiceTest.php:34

Caused by
Doctrine\DBAL\Driver\Mysqli\MysqliException: No such file or directory

/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php:73
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/Driver.php:41
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:64
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:173
/Volumes/Projects/MyProject/Web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1398
/Volumes/Projects/MyProject/Web/vendor/typo3/testing-framework/Classes/Core/Testbase.php:503
/Volumes/Projects/MyProject/Web/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:274
/Volumes/Projects/MyProject/Web/typo3conf/ext/acme_base/Tests/Functional/Service/SettingsServiceTest.php:34

...
```

**Comment by timohund** [comment]
Did you set TYPO3_PATH_PACKAGES in the build process? That's what i need todo today to get our build (EXT:solr) running with the latest changes in the master branch:

 https://github.com/TYPO3-Solr/ext-solr/pull/1227/files

**Comment by markwatney2016** [comment]
Are you sure that this is still relevant? https://github.com/TYPO3/TYPO3.CMS/commit/dc824fec9431224d8f2305622232bf871a8763a2

**Comment by markwatney2016** [comment]
I could reproduce the reported problem with a fresh TYPO3 instance doing this:

1. Installing TYPO3

```
mkdir foo && cd foo \
&& \
composer require typo3/cms \
&& \
# We want to execute TYPO3's functional tests so we delete TYPO3 and install it's sources
unlink index.php && unlink typo3 && rm -rf typo3_src \
&& \
composer install --prefer-source \
&& \
composer require typo3/testing-framework \
&& \
composer require helhum/typo3-console \
&& \
./vendor/bin/typo3cms install:setup \
&& \
./vendor/bin/typo3cms install:generatepackagestates \
&& \
./vendor/bin/typo3cms install:fixfolderstructure \
```

2. Adding some outoloading configuration to `composer.json`:

```
    "autoload-dev": {
        "psr-4": {
          "TYPO3\\CMS\\Backend\\Tests\\": "typo3/sysext/backend/Tests/",
          "TYPO3\\CMS\\Belog\\Tests\\": "typo3/sysext/belog/Tests/",
          "TYPO3\\CMS\\Beuser\\Tests\\": "typo3/sysext/beuser/Tests/",
          "TYPO3\\CMS\\Core\\Tests\\": "typo3/sysext/core/Tests/",
          "TYPO3\\CMS\\Documentation\\Tests\\": "typo3/sysext/documentation/Tests/",
          "TYPO3\\CMS\\Extbase\\Tests\\": "typo3/sysext/extbase/Tests/",
          "TYPO3\\CMS\\Extensionmanager\\Tests\\": "typo3/sysext/extensionmanager/Tests/",
          "TYPO3\\CMS\\Felogin\\Tests\\": "typo3/sysext/felogin/Tests/",
          "TYPO3\\CMS\\Filemetadata\\Tests\\": "typo3/sysext/filemetadata/Tests/",
          "TYPO3\\CMS\\Fluid\\Tests\\": "typo3/sysext/fluid/Tests/",
          "TYPO3\\CMS\\FluidStyledContent\\Tests\\": "typo3/sysext/fluid_styled_content/Tests/",
          "TYPO3\\CMS\\Form\\Tests\\": "typo3/sysext/form/Tests/",
          "TYPO3\\CMS\\Frontend\\Tests\\": "typo3/sysext/frontend/Tests/",
          "TYPO3\\CMS\\Impexp\\Tests\\": "typo3/sysext/impexp/Tests/",
          "TYPO3\\CMS\\IndexedSearch\\Tests\\": "typo3/sysext/indexed_search/Tests/",
          "TYPO3\\CMS\\Install\\Tests\\": "typo3/sysext/install/Tests/",
          "TYPO3\\CMS\\Lang\\Tests\\": "typo3/sysext/lang/Tests/",
          "TYPO3\\CMS\\Lowlevel\\Tests\\": "typo3/sysext/lowlevel/Tests/",
          "TYPO3\\CMS\\Recordlist\\Tests\\": "typo3/sysext/recordlist/Tests/",
          "TYPO3\\CMS\\Reports\\Tests\\": "typo3/sysext/reports/Tests/",
          "TYPO3\\CMS\\Rsaauth\\Tests\\": "typo3/sysext/rsaauth/Tests/",
          "TYPO3\\CMS\\Saltedpasswords\\Tests\\": "typo3/sysext/saltedpasswords/Tests/",
          "TYPO3\\CMS\\Scheduler\\Tests\\": "typo3/sysext/scheduler/Tests/",
          "TYPO3\\CMS\\Sv\\Tests\\": "typo3/sysext/sv/Tests/",
          "TYPO3\\CMS\\SysAction\\Tests\\": "typo3/sysext/sys_action/Tests/",
          "TYPO3\\CMS\\SysNote\\Tests\\": "typo3/sysext/sys_note/Tests/",
          "TYPO3\\CMS\\Typo3DbLegacy\\Tests\\": "typo3/sysext/typo3db_legacy/Tests/",
          "TYPO3\\CMS\\Workspaces\\Tests\\": "typo3/sysext/workspaces/Tests/",
          "TYPO3\\CMS\\Recycler\\Tests\\": "typo3/sysext/recycler/Tests/"
        },
        "classmap": ["typo3/sysext/extbase/Tests/Unit/Object/Container/Fixtures/"]
      }
```

3. Running a Functional Test of TYPO3

```
typo3DatabaseName="functional_test" typo3DatabaseUsername="root" typo3DatabasePassword="root" typo3DatabaseHost="127.0.0.1" ./vendor/bin/phpunit -c ./vendor/typo3/testing-framework/Resources/Core/Build/FunctionalTests.xml
```

This error is thrown:

```
➜  foo typo3DatabaseName="functional_test" typo3DatabaseUsername="root" typo3DatabasePassword="root" typo3DatabaseHost="127.0.0.1" ./vendor/bin/phpunit -c ./vendor/typo3/testing-framework/Resources/Core/Build/FunctionalTests.xml ./typo3/sysext/core/Tests/Functional/Log/Writer/DatabaseWriterTest.php
PHPUnit 5.7.20 by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 180 ms, Memory: 4.00MB

There was 1 error:

1) TYPO3\CMS\Core\Tests\Functional\Log\Writer\DatabaseWriterTest::writeLogInsertsLogRecordWithGivenProperties
Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: No such file or directory

/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:103
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:145
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/Driver.php:38
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:64
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:173
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1398
/Users/markwatney/__temp/foo/vendor/typo3/testing-framework/Classes/Core/Testbase.php:502
/Users/markwatney/__temp/foo/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:276

Caused by
Doctrine\DBAL\Driver\Mysqli\MysqliException: No such file or directory

/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php:73
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/Driver.php:36
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:64
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:173
/Users/markwatney/__temp/foo/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1398
/Users/markwatney/__temp/foo/vendor/typo3/testing-framework/Classes/Core/Testbase.php:502
/Users/markwatney/__temp/foo/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:276

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

**Comment by markwatney2016** [comment]
The solution is most probably somethink like this adjustment just made in `nimut/testing-framework`: https://github.com/Nimut/testing-framework/commit/3e6ae9bb1497f907e3481bf9a326d0d1c0db3d34

Further details can be found here (and in it's previous posts): https://typo3.slack.com/archives/C027S5XR1/p1495612123797825

**Comment by psychomieze** [comment]
@markwatney2016 Can you try adding this line in your FunctionalTestCase.php->setUp (ll221) and report if that fixes your problem?
putenv('TYPO3_PATH_ROOT=' . $this->instancePath);

**Comment by lolli42** [comment]
this should be fixed with eafaac680acc34e6288a50d90cecc1d02dfb6e66
